### ISSUE TITLE
Add lr_scheduler options in BaseCV.configure_optimizers

### ISSUE THREAD

**Issue by EnricoTrizio** [luigibonati/mlcolvar#96]
The implementation of the configure_optimizer method of BaseCV apparently doesn't allow the use of a lr_scheduler.

This could be included in the 'optimizer' options maybe, for example, in a quick and dirty way

```
def configure_optimizers(self):
        """
        Initialize the optimizer based on self._optimizer_name and self.optimizer_kwargs.

        Returns
        -------
        torch.optim
            Torch optimizer
        """
        lr_scheduler_dict = self.optimizer_kwargs.pop('lr_scheduler', None)
    
        optimizer = getattr(torch.optim, self._optimizer_name)(
            self.parameters(), **self.optimizer_kwargs
        )
        if lr_scheduler_dict is not None:
            lr_scheduler_name = lr_scheduler_dict.pop('scheduler')
            lr_scheduler = {
                'scheduler': lr_scheduler_name(optimizer, **lr_scheduler_dict),
            }
            lr_scheduler_dict['scheduler'] = lr_scheduler_name
            self.optimizer_kwargs['lr_scheduler'] = lr_scheduler_dict
            return [optimizer] , [lr_scheduler]
        else: 
            return optimizer
```

**Comment by EnricoTrizio** [comment]
A minimal usage example following the strategy of the previous comment.
The scheduler and its options are passed as options of the optimizer.

One of the problems is that the optimizer needs to be initialized before the scheduler and I would also avoid having to make a matryoshka of dictionaries.

```
lr_scheduler = torch.optim.lr_scheduler.ExponentialLR
options = {'optimizer' : {'lr': 1e-3, 'lr_scheduler' : { 'scheduler' : lr_scheduler, 'gamma' : 0.99999 } } }

model = CV_model(layers = [2, 32, 32, 1], options=options)
```
